supabase/migrations/20240316051441_tables_add_threads_and_messages..sql

```diff
@@ -0,0 +1,39 @@
+CREATE TABLE IF NOT EXISTS threads (
+  id uuid NOT NULL PRIMARY KEY,
+  service_provider_id uuid NOT NULL,
+  customer_id uuid NOT NULL,
+  created_at timestamptz NOT NULL DEFAULT NOW(),
+  updated_at timestamptz NOT NULL DEFAULT NOW(),
+
+  FOREIGN KEY (service_provider_id) REFERENCES service_providers(user_id),
+  FOREIGN KEY (customer_id) REFERENCES customers(user_id)
+);
+
+ALTER TABLE threads
+  ENABLE ROW LEVEL SECURITY;
+
+CREATE POLICY "Can view own customer threads" ON threads
+  FOR SELECT USING (auth.uid() = customer_id);
+CREATE POLICY "Can update own service provider threads" ON threads
+  FOR UPDATE USING (auth.uid() = service_provider_id);
+
+CREATE TABLE IF NOT EXISTS messages (
+  id serial NOT NULL PRIMARY KEY,
```

Comment:
This definitely a place where I'd prefer some sort of Ordered ID like UUID7 (which has a time-based component) over fully random IDs. Having IDs that are ordered increases the chances that different rows are clustered together, thereby improving write performance since the database can batch operations better.

For now, I'll just go with the serial auto-incrementing integer data type. Probably good enough for now.

**Resources:**
Shopify Blog Post - Heading 6. Use Idempotency Keys
https://shopify.engineering/building-resilient-payment-systems

Breakdown by Hussein Nasser
https://www.youtube.com/watch?v=f53-Iw_5ucA